internal/source/git.go

```diff
@@ -231,10 +237,26 @@ func (f *billyFS) ReadDir(name string) ([]fs.DirEntry, error) {
 
 type billyFile struct {
 	billy.File
-	fi    os.FileInfo
-	fiErr error
+	fi os.FileInfo
 }
 
 func (b billyFile) Stat() (fs.FileInfo, error) {
-	return b.fi, b.fiErr
+	return b.fi, nil
+}
+
+func (b billyFile) Close() error {
+	if b.File != nil {
+		return b.File.Close()
+	}
+	return nil
+}
+
+type billyDirFile struct {
+	billyFile
+	fs   *billyFS
+	path string
+}
+
+func (d *billyDirFile) ReadDir(n int) ([]fs.DirEntry, error) {
+	return d.fs.ReadDir(d.path)
```

Comment:
The [docs](https://pkg.go.dev/io/fs#ReadDirFile) for implementing this function are pretty confusing to me, specifically:
> 	// In [the case of n > 0], if ReadDir returns an empty slice, it will return
	// a non-nil error explaining why.
	// At the end of a directory, the error is io.EOF.
	// (ReadDir must return io.EOF itself, not an error wrapping io.EOF.)

I assume we can rely on `b.fs.ReadDir` to handle those specifics, but I'd suggest something like this, so we at least honor the `n` parameter and follow the other piece of the ReadDirFile docs that talk about returning a partial list in the event of an error.
```go
	entries, err := b.fs.ReadDir(b.path)
	if n <= 0 {
		return entries, err
	}
	if n > len(entries) {
		n = len(entries)
	}
	return entries[:n], err
```